zio/detector/file.go

```diff
@@ -31,6 +31,10 @@ const StdinPath = "/dev/stdin"
 // URL. If the path is neither of these or can't otherwise be opened,
 // an error is returned.
 func OpenFile(zctx *resolver.Context, path string, opts zio.ReaderOpts) (*zbuf.File, error) {
+	return OpenFileWithContext(context.Background(), zctx, path, opts)
+}
+
+func OpenFileWithContext(ctx context.Context, zctx *resolver.Context, path string, opts zio.ReaderOpts) (*zbuf.File, error) {
```

Comment:
Nit: Maybe add the context parameter to OpenFile and other existing constructors instead of adding WithContext variants?

Reply:
I went back and forth on this. I suppose it's a matter of taste, but all the `context.Background()` calls for simple command line tool and tests got to me.

Reply:
I get it.